### ISSUE TITLE
`<bit>`: Implement `rotl()` and `rotr()` with intrinsics

### ISSUE THREAD

**Issue by StephanTLavavej** [microsoft/STL#2367]
https://github.com/microsoft/STL/blob/3c2fd04d441d46ec9d914d9cbb621a3bac96c3a5/stl/inc/bit#L92-L118

`intrin.h` declares:

```cpp
__MACHINE(unsigned char __cdecl _rotl8(unsigned char _Value, unsigned char _Shift))
__MACHINE(unsigned short __cdecl _rotl16(unsigned short _Value, unsigned char _Shift))
__MACHINE(unsigned int __cdecl _rotl(_In_ unsigned int _Value, _In_ int _Shift))
__MACHINE(unsigned __int64 __cdecl _rotl64(_In_ unsigned __int64 _Value, _In_ int _Shift))

__MACHINE(unsigned char __cdecl _rotr8(unsigned char _Value, unsigned char _Shift))
__MACHINE(unsigned short __cdecl _rotr16(unsigned short _Value, unsigned char _Shift))
__MACHINE(unsigned int __cdecl _rotr(_In_ unsigned int _Value, _In_ int _Shift))
__MACHINE(unsigned __int64 __cdecl _rotr64(_In_ unsigned __int64 _Value, _In_ int _Shift))
```

* We'll need to move these to `intrin0.h` (physically, `intrin0.inl.h`) if we want throughput.
* We'll need to make sure this works for Clang.
* `<bit>` is C++20, so we don't need to worry about `/clr:pure` (for sure), CUDA (for sure), or the Intel C++ Compiler (I think).

Originally reported as DevCom-1590917 and VSO-1440998 / AB#1440998 .

**Comment by AlexGuteniev** [comment]
> We'll need to move these to intrin0.h (physically, intrin0.inl.h) if we want throughput.

`_rotl` is in `<stdlib.h>` as per [documentation](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/rotl-rotl64-rotr-rotr64?view=msvc-170), I confirm it is in the SDK, can we rely on that instead of having it in `<intrin0.h>` ?

I suspect yes, as the same was true for `_byteswap_ulong` et al.

**Comment by AlexGuteniev** [comment]
Oh, actually not fully useful as there's no 8 bit and 16 bit versions

**Comment by StephanTLavavej** [comment]
Reopening as the 8-bit and 16-bit cases remain to be done. Internal MSVC-PR-368557 has moved those intrinsics to `intrin0.h` and will be available in VS 2022 17.2 Preview 2.

**Comment by StephanTLavavej** [comment]
The 8-bit and 16-bit cases were completed by #2611.
